applications/GeoMechanicsApplication/custom_conditions/U_Pw_normal_lysmer_absorbing_condition.cpp

```diff
@@ -25,190 +25,218 @@ Condition::Pointer UPwLysmerAbsorbingCondition<TDim,TNumNodes>::Create(IndexType
+}
 
-    // assemble left hand side vector
-    if (rLhsMatrix.size1() != CONDITION_SIZE)
-        rLhsMatrix.resize(CONDITION_SIZE, CONDITION_SIZE, false);
 
-    noalias(rLhsMatrix) = ZeroMatrix(CONDITION_SIZE, CONDITION_SIZE);
-    this->CalculateAndAddLHS(rLhsMatrix, rVariables);
+template< unsigned int TDim, unsigned int TNumNodes >
+void UPwLysmerAbsorbingCondition<TDim, TNumNodes>::CalculateRightHandSide(VectorType& rRightHandSideVector,
+    const ProcessInfo& rCurrentProcessInfo)
+{
+    BoundedMatrix<double, N_DOF, N_DOF> stiffness_matrix;
 
-    // no righthand side contribution
-    if (rRightHandSideVector.size() != CONDITION_SIZE)
-        rRightHandSideVector.resize(CONDITION_SIZE, false);
-    noalias(rRightHandSideVector) = ZeroVector(CONDITION_SIZE);
+    this->CalculateConditionStiffnessMatrix(stiffness_matrix, rCurrentProcessInfo);
+
+    MatrixType global_stiffness_matrix = ZeroMatrix(CONDITION_SIZE, CONDITION_SIZE);
+    GeoElementUtilities::AssembleUBlockMatrix< TDim, TNumNodes >(global_stiffness_matrix, stiffness_matrix);
+
+    this->CalculateAndAddRHS(rRightHandSideVector, global_stiffness_matrix);
 }
 
-/// <summary>
-/// Calculates LHS Damping part of abosrbing boundary
-/// </summary>
-/// <param name="rDampingMatrix"></param>
-/// <param name="CurrentProcessInfo"></param>
+
 template< unsigned int TDim, unsigned int TNumNodes >
-void UPwLysmerAbsorbingCondition<TDim, TNumNodes>::CalculateDampingMatrix(MatrixType& rDampingMatrix, const ProcessInfo& CurrentProcessInfo)
+void UPwLysmerAbsorbingCondition<TDim, TNumNodes>::CalculateDampingMatrix(MatrixType& rDampingMatrix, const ProcessInfo& rCurrentProcessInfo)
 {
 
     //Previous definitions
-    GeometryType& rGeom = this->GetGeometry();
+    GeometryType& r_geom = this->GetGeometry();
 
-    GeometryData::IntegrationMethod rIntegrationMethod = this->mThisIntegrationMethod;
-    const GeometryType::IntegrationPointsArrayType& IntegrationPoints = rGeom.IntegrationPoints(rIntegrationMethod);
-    const unsigned int NumGPoints = IntegrationPoints.size();
-    const unsigned int LocalDim = rGeom.LocalSpaceDimension();
+    GeometryData::IntegrationMethod r_integration_method = this->mThisIntegrationMethod;
+    const GeometryType::IntegrationPointsArrayType& r_integration_points = r_geom.IntegrationPoints(r_integration_method);
+    const unsigned int num_g_points = r_integration_points.size();
+    const unsigned int local_dim = r_geom.LocalSpaceDimension();
 
     //Containers of variables at all integration points
-    const Matrix& NContainer = rGeom.ShapeFunctionsValues(rIntegrationMethod);
-    GeometryType::JacobiansType JContainer(NumGPoints);
-    for (unsigned int i = 0; i < NumGPoints; ++i)
-        (JContainer[i]).resize(TDim, LocalDim, false);
-    rGeom.Jacobian(JContainer, rIntegrationMethod);
+    const Matrix& r_n_container = r_geom.ShapeFunctionsValues(r_integration_method);
+    GeometryType::JacobiansType j_container(num_g_points);
```

Comment:
If you also like `jacobians` more than `j_container`, please update it here too:
```suggestion
    GeometryType::JacobiansType jacobians(num_g_points);
```